plugin/EstateList.php

```diff
@@ -656,6 +661,89 @@ public function getEstateIds(): array
 		return array_column($this->_records, 'id');
 	}
 
+	/**
+	 * @param array $parkingArray
+	 * @param string $language
+	 * @param string $locate
+	 * @return array
+	 */
+	public function formatParkingLot(array $parkingArray, string $language, string $locale = 'de_DE'): array
+	{
+		$messages = [];
+		foreach ($parkingArray as $key => $parking) {
+			if (!$parking['Count']) {
+				continue;
+			}
+			switch ($language) {
+				case 'ENG':
+					$element = sprintf("%d %s at %s", number_format_i18n($parking['Count']), $this->getParkingName($key, $parking['Count']), $this->formatPrice($parking['Price'], $language, $locale));
+					break;
+				default:
+					$element = sprintf("%d %s à %s", number_format_i18n($parking['Count']), $this->getParkingName($key, $parking['Count']), $this->formatPrice($parking['Price'], $language, $locale));
+					break;
+			}
+			if (!empty($parking['MarketingType'])) {
+				$element .= " ({$parking['MarketingType']})";
+			}
+			array_push($messages, $element);
+		}
+		return $messages;
+	}
+
+	/**
+	 * @param string $str
+	 * @param string $language
+	 * @param string $locate
+	 * @return string
+	 */
+	public function formatPrice(string $str, string $language, string $locale): string
+	{
+		$digit = intval(substr(strrchr($str, "."), 1));
+		try {
+			$format = new NumberFormatter($locale, NumberFormatter::CURRENCY);
+			if ($digit) {
+				$format->setAttribute(NumberFormatter::MAX_FRACTION_DIGITS, 2);
+			} else {
+				$format->setAttribute(NumberFormatter::MAX_FRACTION_DIGITS, 0);
+			}
+			return str_replace("\xc2\xa0", " ", $format->formatCurrency($str, "EUR"));;
+		} catch (\Exception $exception){
+			if ($digit) {
+				$str = floatval($str);
+				$str = number_format_i18n($str, 2);
+			} else {
+				$str = number_format_i18n(intval($str));
+			}
+			switch ($language) {
+				case 'ENG':
+					$str = '€' . $str;
+					break;
+				default:
+					$str = $str . ' €';
+					break;
+			}
+			return $str;
+		}
+	}
+
+	/**
+	 * @param string $parkingName
+	 * @param int $count
+	 * @return string
+	 */
+	public function getParkingName(string $parkingName, int $count): string
+	{
+		$str = preg_replace('/([a-z])([A-Z])/', "\\1 \\2", $parkingName);
+		switch ($count) {
```

Comment:
The `_n()` function uses the singular if `$count` is `1` and the plural otherwise. There are languages with different rules and `_n()` respects those.

Reply:
The switch/case should look like this in order to be properly translated:
```php
switch ($parkingName) {
    case 'carport':
        /* translators: %s is the amount of carports */
        $str = _n('%s carport', '%s carports', $count, 'onoffice-for-wp-websites');
        break;
    case 'duplex':
         /* translators: %s is the amount of duplexes */
        $str = _n('%s duplex', '%s duplexes', $count, 'onoffice-for-wp-websites');
        break;
     ....
}
return esc_html(sprintf($str, $count));
```